cpp/src/arrow/CMakeLists.txt

```diff
@@ -143,14 +143,19 @@ if(ARROW_COMPUTE)
   set(ARROW_SRCS
       ${ARROW_SRCS}
       compute/context.cc
+      compute/expression.cc
```

Comment:
Are You that this should be organized under compute? I think We should separate it.

Compute does actual computation eagerly, whereas `expression` and `operations` modules are defining a logical plan, which ideally should be optimized before being passed to the execution layer. The execution layer could be `arrow::compute` or `gandiva`or even another pluggable one.

Reply:
My intent is to create `arrow/compute/execution` directory